core/Test/Tasty/Run.hs

```diff
@@ -171,7 +177,9 @@ executeTest action statusVar timeoutOpt inits fins = mask $ \restore -> do
             , resultTime = fromIntegral t
             , resultDetailsPrinter = noResultDetails
             }
-      fromMaybe timeoutResult <$> timeout t a
+      -- If compiled with unbounded-delays then t' :: Integer, otherwise t' :: Int
+      let t' = fromInteger (min (max 0 t) (toInteger (maxBound :: Int64)))
```

Comment:
Is the `max 0 t` necessary? Both [`System.Timeout.timeout`](https://hackage.haskell.org/package/base-4.8.2.0/docs/src/System.Timeout.html#timeout) and its [`unbounded-delays` counterpart](https://hackage.haskell.org/package/unbounded-delays-0.1.1.1/docs/src/Control.Concurrent.Timeout.html#timeoutWithPred) are `fmap Just` when the specified delay is less than zero.

That being said, it doesn't hurt to keep it there. :)

Reply:
`max 0 t` is to prevent underflow. Imagine `t = -(2^63+1) :: Integer`: previously it meant no delay, but without `max 0 t` it would be cast to `2^63-1 :: Int`, a huge timeout.